.github/workflows/ci.yml

```diff
@@ -0,0 +1,111 @@
+name: Pipeline
+
+on:
+  push:
+    branches:
+      - test_task
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+    inputs:
+      registry:
+        description: "Registry"
+        required: false
+        default: "ghcr.io"
+        type: string
+      server_address:
+        description: "Server address"
+        required: false
+        default: "smtp.gmail.com"
+        type: string
+
+env:
+  GH_TOKEN: ${{ github.token }}
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ github.workspace }}/angular
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: start timer
+        run: |
+          echo "START_TIME=$(date +%s)" >> $GITHUB_ENV
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version-file: "angular/package.json"
+
+      - name: install dependencies
+        run: npm ci
+
+      - name: build /dist
+        run: npm run build
+
+      #- name: run tests
+      #  run: npm run coverage
+
+      - name: login to github container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ inputs.registry }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.PAT }}
+
+      - name: extract metadata (tags, labels) for docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+        with:
+          images: ${{ inputs.registry }}/${{ github.repository }}
+
+      - name: build and push docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
+        with:
+          context: .
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+
+      - name: stop timer
+        if: always()
+        run: |
+          echo "STOP_TIME=$(date +%s)" >> $GITHUB_ENV
+
+      - name: counting the time
+        if: always()
+        run: |
```

Comment:
This is too complicated  implementation for such a simple logic, pls find out a way to make it one liner